tests/drivers/gpio/gpio_basic_api/src/gpio_emul_callback.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_GPIO_EMUL
```

Comment:
Can this move to the CMakeLists.txt file?

Reply:
I think that should work - let me give it a shot.

Reply:
Looks like it won't due to globbing.
```
FILE(GLOB app_sources src/*.c)
```

Reply:
Change `CMakeLists.txt`:

```
FILE(GLOB app_sources src/main.c src/test*.c)
target_sources(app PRIVATE ${app_sources})
target_sources_ifdef(CONFIG_GPIO_EMUL app PRIVATE src/gpio_emul_callback.c)
```
and get rid of the whole-file `#if/#endif`.